### ISSUE TITLE
F-Droid

### ISSUE THREAD

**Issue by qoheniac** [ProtonMail/proton-mail-android#1]
Thank you for sharing the source code of your Android app! I'm just curious whether it is planned to add the app to the F-Droid repository now.

**Comment by waltercool** [comment]
Hope the F-Droid doesn't make so much complicated like it been happening for ProtoVPN. Been following that thread, and looks like it takes forever for them to answer.

**Comment by JarlavGrenland** [comment]
Maybe Proton should make their own repo where they will be able to push updates to both Mail and VPN 🤔

**Comment by waltercool** [comment]
That's an option, just like what MicroG does for themselves https://microg.org/download.html

**Comment by qoheniac** [comment]
I think the discussions mentioned here are [this](https://github.com/ProtonVPN/android-app/issues/1) and [this](https://gitlab.com/fdroid/rfp/issues/1206).

**Comment by Braincoke** [comment]
It appears that it is part of their roadmap from a [question asked](https://gitlab.com/fdroid/rfp/issues/1206#note_305521602) in the ProtonVPN Request For Packaging.

**Comment by 9034725985** [comment]
My understanding is this is blocked by https://github.com/ProtonMail/proton-mail-android/issues/3 

needs to build before it can be in f-droid

**Comment by jpellegrini** [comment]
> My understanding is this is blocked by #3
> 
> needs to build before it can be in f-droid

That makes sense... If I can't build it myself, should I consider it free, or open-source? The source code must correspond to the package I use, and -- I should be able to build it myself.

**Comment by licaon-kter** [comment]
It's not like every F-Droid volunteer is suddenly familiar with insert-random-app build environment and build tools. :)

Anyone can just pop and leave a MR for the recipe that compiles the needed libs so we can just check it and merge it, right? ;)

**Comment by licaon-kter** [comment]
In the mean time, can we have Tags?

And maybe some sort of folder that contains the app metadata _(summary/description/images/changelog/etc)_ in a [Fastlane](https://gitlab.com/snippets/1895688) or [Triple-T](https://gitlab.com/snippets/1901490) structure?

Also stuff like source-code for [these libs here](https://github.com/ProtonMail/proton-mail-android/tree/release/app/libs) _(are these in [a trusted by F-Droid](https://gitlab.com/fdroid/fdroidserver/-/blob/master/fdroidserver/scanner.py#L100) maven repo maybe?)_ and surely a Gradle flavour that excludes all the non-FOSS libs like GMS and whatnot :)

**Comment by m-boyd** [comment]
Request for packaging in F-Droid: 
https://gitlab.com/fdroid/rfp/issues/1323

**Comment by m-boyd** [comment]
News from [IzzySoft](https://gitlab.com/fdroid/rfp/-/issues/1323#note_331475606): 

> Well, unfortunately…
>```
>implementation 'com.google.android.gms:play-services-gcm:17.0.0'
>implementation 'com.google.android.gms:play-services-safetynet:17.0.0'
>```
>And probably some more. So they need to be convinced to either kick that altogether – or add a build flavor that skips them.

@ZoricaStojchevska, can you help with this?

**Comment by m-boyd** [comment]
[Results from fdroid-bot](https://gitlab.com/fdroid/rfp/-/issues/1323#note_331480320): 

>- found https://play.google.com/store/apps/details?id=ch.protonmail.android
>- _gradle/wrapper/gradle-wrapper.jar_ does not match the wrapper version (https://services.gradle.org/distributions/gradle-6.3-all.zip).  See update and verify instructions: https://docs.gradle.org/current/userguide/gradle_wrapper.html#sec:upgrading_wrapper
>
>### fdroid scanner
>https://github.com/ProtonMail/proton-mail-android:
>```
>$ fdroid scanner --verbose ch.protonmail.android
>DEBUG: Reading 'config.py'
>INFO: ch.protonmail.android: no builds specified, running on current source state
>WARNING: Found JAR file at gopenpgp/gopenpgp-sources.jar
>WARNING: Found AAR file at gopenpgp/gopenpgp.aar
>INFO: Removing gradle-wrapper.jar at gradle/wrapper/gradle-wrapper.jar
>INFO: Removing gradle-wrapper.jar at gradle/gradle/wrapper/gradle-wrapper.jar
>ERROR: Found usual suspect 'google.*play.*services' at line 132 at app/build.gradle
>ERROR: Found usual suspect 'google.*play.*services' at line 150 at app/build.gradle
>WARNING: Found JAR file at app/libs/remark-1.1.0.jar
>WARNING: Found JAR file at app/libs/mail-activation.jar
>WARNING: Found JAR file at app/libs/commons-email-1.3.jar
>WARNING: Found JAR file at app/libs/mail.jar
>WARNING: Found AAR file at app/libs/Proton-core_0.2.aar
>WARNING: Found JAR file at app/libs/mail-additional.jar
>WARNING: Found AAR file at app/libs/Proton-core_0.2.15.aar
>WARNING: Found JAR file at app/libs/PM-sources.jar
>WARNING: Found possible binary at app/src/main/assets/pgp-icons-android.ttf
>WARNING: Scanner found 2 problems in ch.protonmail.android:
>INFO: Finished
>2 problems found
>```
>
>### Security Issues
>- Custom _gradle/wrapper/gradle-wrapper.jar_, not found in https://gradle.org/release-checksums: e2b82129ab64751fd40437007bd2f7f2afb3c6e41a9198e628650b22d5824a14
>- _gradle/wrapper/gradle-wrapper.properties_ is missing [distributionSha256Sum](https://docs.gradle.org/current/userguide/gradle_wrapper.html#sec:verification), unverified gradle download! Here is an example of how to fix this:
>
>```
>#Mon Oct 01 11:02:58 CEST 2018
>distributionBase=GRADLE_USER_HOME
>distributionPath=wrapper/dists
>zipStoreBase=GRADLE_USER_HOME
>zipStorePath=wrapper/dists
>distributionUrl=https\://services.gradle.org/distributions/gradle-6.3-all.zip
>distributionSha256Sum=0f316a67b971b7b571dac7215dcf2591a30994b3450e0629925ffcfe2c68cc5c
>```
>
>### External Scanners
>- [APK Scan ch.protonmail.android](https://apkscan.org/?searchby=pkg&q=ch.protonmail.android)

**Comment by hushrom** [comment]
I have an update on ProtonMail's side. Disclaimer: _I'm not a ProtonMail developer or employee_, I just received an update from them. They are currently working on developing their own generic push-notification framework and by doing so removes the need for GCM and firebase so that their services could work throughout all multiple platforms of Android. Unfortunately, there isn't an ETA on when they will finish the project.

**Comment by licaon-kter** [comment]
@hushrom It took them ~5years to open source this, so I won't hold my breath. :)

**Comment by 4face-studi0** [comment]
@m-boyd thank you, probably this happened once we moved to a new GitLab repo - before the open sourcing - and had to solve some conflicts porting branches from old repo.
We're gonna fix that ASAP

**Comment by licaon-kter** [comment]
Look at that recipe...daaaaaamn: https://gitlab.com/fdroid/fdroiddata/-/commit/28339478815bf489b7855fb1f08f5cddc25ac699 :) :)

**Comment by m-boyd** [comment]
@4face-studi0, will ProtonMail produce a build that omits Google Play Services? Or how is this being approached?

@licaon-kter, can you weigh in on [the warnings from f-droid-bot](https://gitlab.com/fdroid/rfp/-/issues/1323#note_331480320)?

**Comment by 4face-studi0** [comment]
> @4face-studi0, will ProtonMail produce a build that omits Google Play Services? Or how is this being approached?
> 
> @licaon-kter, can you weigh in on [the warnings from f-droid-bot](https://gitlab.com/fdroid/rfp/-/issues/1323#note_331480320)?

I could not tell, I don't any information about our roadmap regarding F-Droid and I'm not very familiar with it myself. 
Probably we're goin to exclude Play Services from the Gradle configuration whenever the API key is not provided.
Technically talking 
```kotlin
if (hasGoogleKey()) {
    implementation("url:to:lib")
}
```

**Comment by theo-dep** [comment]
Hey, I'm using LineageOS and ProtonMail but without Google Play services, I don't have notification push. All notification system is based on Google et must be changed before the add into F-Droid.

**Comment by licaon-kter** [comment]
A simple "background check every X mins" could be implemented. _(let's not start on how IMAP IDLE just works without any push 🙄 )_

**Comment by qoheniac** [comment]
[This article](https://f-droid.org/en/2018/09/03/replacing-gcm-in-tutanota.html) describes how Tutanota solved the problem in 2018! But I think it would be better to develop general solutions like [OpenPush](https://f-droid.org/en/2020/02/03/openpush-talk.html) instead of solving it for every single app.

**Comment by ddreth** [comment]
would love to see this happen as well.

**Comment by IzzySoft** [comment]
Until this gets sorted out (which I expect will take a while), I'm taking this app into my repo. It will [show up here](https://apt.izzysoft.de/fdroid/index/apk/ch.protonmail.android) in about an hour – and auto-update whenever a new tag is created and an APK attached to it. Should Fastlane be established (see above), please let me know so I can replace the screenshots (grabbed from Play for now) and integrate per-release changelogs (if provided).

**Comment by IzzySoft** [comment]
Adding FCM with the latest release wasn't really a wise move concerning the issue here. I hope you have plans for a flavor to come without that. Convenience shouldn't weight heigher than privacy.

If you don't mind the pun here: this was a notification pushing the realization further back :cry: In my repo, I just had to add a second AntiFeature hence (NonFreeNet). Two such AntiFeatures do not really look inviting to the privacy minded…

**Comment by icfmp** [comment]
Sad ... I suscribed to Protonmail Plus the day Protonmail source code was opened.

I guess I'll wait for Tutanota to develop their mail import feature then i'll leave protonmail ...

**Comment by 4face-studi0** [comment]
@IzzySoft FCM has not been *added*. It just replaced the deprecated *gcm*

**Comment by IzzySoft** [comment]
Still, if I'm looking for a privacy-focused mail service and the app greets me with (currently) 3 AntiFeatures (NonFreeDep because of GMS, NonFreeNet because of Firebase, and Tracking because of Sentry – though the latter might be disabled by default, which is not clear by the app's description) that rather scares me away. I'd prefer it fully libre (opt-in Sentry should be fine as long as it's self-hosted – but GMS/Firebase are IMHO not and definitely show-stopper for F-Droid). No accusations or bad feelings, just stating the facts – I'm glad services like ProtonMail exist!

**Comment by 4face-studi0** [comment]
@IzzySoft other team are working for bring a Google free notification system, but not being there, as now there are no other choices as:
1) it's the only reliable and optimized notification system 
2) we were already using it and the migration away from the deprecated one was necessary at this point.

Some compromises are necessary, information sent via FCM are the bare minimum for being able to retrieve and decrypt the message after hand.
Sentry as said is a self-hosted and is necessary for guarantee a stability of the software.

( My 2 cents: if you have Play Services in your device, having FCM or not in your applications doesn't matter as the library will only interface with the system; if you don't have it, the library just won't work and the app will tell you that Play Services are not available )

**Comment by licaon-kter** [comment]
@4face-studi0 it's a closed source lib, how do you know what it does or not?

**Comment by IzzySoft** [comment]
> it's the only reliable and optimized notification system

And is needed for … what? I use k-9 mail, no such thing in there. I'm fine with an interval of 1h for "polling" – there's nothing *that* urgent. Anf for those where it is, there's "IMAP idle". So I don't get why each flashlight app needs FCM (sorry).

> Some compromises are necessary,

Are they? Maybe, but I don't see why this specific one should.

> if you don't have it, the library just won't work and the app will tell you that Play Services are not available

My 2 cent: 1) that proves the library is not really needed, and 2) as @licaon-kter just said: did you see the source? How can you be so sure what a blob does and what not?

**Comment by JulienPalard** [comment]
I'm +1 for the "1h polling".

I'm not saying "drop FCM completly", but at least providing a build flag to switch from FCM to fixed-rate polling would be a step towards an f-droid release. (I would even say: a flag to **enable** FCM, as building without close souce components should be the default behavior).

**Comment by polyzen** [comment]
K-9 might not be the best example:
https://github.com/k9mail/k-9/issues/3950
https://github.com/k9mail/k-9/issues/4258

**Comment by IzzySoft** [comment]
@polyzen take your pick then. Their new betas work fine for me (that I'd wish I could have different accounts sync at different times is something else). If you prefer, use Tutanota as example (they've established a Google independent push more than a year ago).

**Comment by proletarius101** [comment]
Apart from gms, how about those libs: https://github.com/ProtonMail/proton-mail-android/commit/d372c9c4398abf09cfdec4766144f5bfe1509184? Are they actually some version of https://github.com/eclipse-ee4j/mail or will be replaced/open-sourced one day?

**Comment by hefty-fading-culprit-epiphany-drew** [comment]
Any update on this? Tuta and ctemplar are already there, you guys are falling behind

**Comment by 4face-studi0** [comment]
> Apart from gms, how about those libs: [d372c9c](https://github.com/ProtonMail/proton-mail-android/commit/d372c9c4398abf09cfdec4766144f5bfe1509184)? Are they actually some version of https://github.com/eclipse-ee4j/mail or will be replaced/open-sourced one day? @4face-studi0

Sorry for late answer, as you can see from the commit message, we tried to remove them, but we have to revert, as some impedance blocked us. We're cooperating with Core team, in order to provide a replacement

**Comment by julianfairfax** [comment]
Given that multiple other FOSS applications have added their own push notifications implementation, wouldn't it be possible to "inspire yourself from"/copy their implementation with some changes for ProtonMail? This is, after all, an open source email application and one of the only applications on my phone that still isn't in F-Droid, which is disappointing to say the least. Thank you for all the work you've done on this application and hopefully this feature will be implemented soon! (For anyone still wanting to download it, you won't have notifications but you can get it from https://protonapps.com/protonmail-android)

**Comment by IzzySoft** [comment]
If push notifications are really thought to be required, be welcome to also [take a look here](https://gitlab.com/-/snippets/1896503#push) for some alternatives. [Gotify](https://github.com/gotify/server) could be self-hosted, [UnifiedPush](https://unifiedpush.org/) would let the users have their choice.

**Comment by julianfairfax** [comment]
> If push notifications are really thought to be required, be welcome to also [take a look here](https://gitlab.com/-/snippets/1896503#push) for some alternatives. [Gotify](https://github.com/gotify/server) could be self-hosted, [UnifiedPush](https://unifiedpush.org/) would let the users have their choice.
Well I personally don't care but ProtonMail's stance seems to be they won't put it on F-Droid without them and F-Droid won't add ProtonMail with a requirement for microG (I know third-party repos like yours have it but it'd be better if ProtonMail and F-Droid work together to get it on there officially, especially since they seem to be willing, unlike something like Signal)

**Comment by proletarius101** [comment]
> ProtonMail with a requirement for microG

A requirement for MicroG (or any other external components, proprietary or open-source) is not a concern. The thing is, FCM requires ProtonMail to have GMS (a proprietary library) inside it.

**Comment by IzzySoft** [comment]
> especially since they seem to be willing

Definitely. And F-Droid is absolutely willing to accept them (as one of the maintainers, I think I can give this statement semi-officially :smile:). But as @proletarius101 just indicated, we cannot compromise on proprietary components being show-stoppers.

Implementing **[UnifiedPush](https://unifiedpush.org/)** instead would give users the choice to use FCM when available, or Gotify or any of the other supported push services – or none at all (if made optional). A Gotify client is even [available at F-Droid](https://f-droid.org/packages/com.github.gotify/) already, so is [NoProvider2Push](https://f-droid.org/packages/com.flyingpanda.noprovider2push/). UnifiedPush is used by at least 2 apps which are at F-Droid as well: [Fedilab](https://fedilab.app/) and [FluffyChat](https://fluffychat.im/), Element (the Matrix client) seems about to implement it as well.

UnifiedPush seems to be implemented in-app as a library. I'd suggest using that. Why? Because for users of GMS, it would offer FCM support out-of-the-box (no additional app needed). F-Droid users wanting push notifications without GMS/FCM could then install the Gotify or the NoProvider2Push client and choose a proper Gotify backend server. WWW (win-win-win), right? GMS users can keep their FCM, privacy proponents can use Gotify/NoProvider2Push, and ProtonMail can be listed at "F-Droid proper". At least that's the theory. Not being an Android dev myself, all I can do is educated guessing :see_no_evil:

**Comment by julianfairfax** [comment]
If you'd need to install a client to use the notifications I wouldn't see myself doing that. since you mentioned Element (the Matrix client), it does all the notifications within it's own app as fair as I know?

**Comment by ArchangeGabriel** [comment]
> If you'd need to install a client to use the notifications I wouldn't see myself doing that. since you mentioned Element (the Matrix client), it does all the notifications within it's own app as fair as I know?

Not notifications, push notifications. Also you would only need to install the side app once for all applications that will use it.

Regarding Element, currently it does either push using FCM (GooglePlay version) or polling (F-Droid version). The latter increases battery usage, which is what push notifications are trying to solve. So they are going to implement UnifiedPush to solve this issue, and I guess that they will only use it in the F-Droid flavor, not PlayStore one.

**Comment by julianfairfax** [comment]
> > If you'd need to install a client to use the notifications I wouldn't see myself doing that. since you mentioned Element (the Matrix client), it does all the notifications within it's own app as fair as I know?
> 
> Not notifications, push notifications. Also you would only need to install the side app once for all applications that will use it.
> 
> Regarding Element, currently it does either push using FCM (GooglePlay version) or polling (F-Droid version). The latter increases battery usage, which is what push notifications are trying to solve. So they are going to implement UnifiedPush to solve this issue, and I guess that they will only use it in the F-Droid flavor, not PlayStore one.

Oh ok, I'm personally fine with a polling solution, or even no notifications so long as the app is on F-Droid. (Asking whoever works at ProtonMail) Would it be possible to add the app to F-Droid with the Google Play Services requirement removed while you work on a replacement? I'm asking this because don't know how long it's going to take for such a replacement to be developed, this issue has been open for more than a year, and if people (like myself) are so insistent on having it on F-Droid, I think they're fine living without the notifications (they could just install the Google Play Store version if not. Just giving us the choice would be a great starting point while something better is worked on. Thanks for your work on this app!

**Comment by ArchangeGabriel** [comment]
You have to understand that such applications without push notifications might be seen has bad UX for users (delays in retrieving content and/or increase battery consumption), something developers don’t want. Even if you would be fine with it, the availability of the app might bring other users in trying it and find it deceptive, meaning they will turn away from it and likely won’t ever know (and thus retry) when push notifications will be available.

Also not everyone understand that this difference exists between F-Droid and PlayStore version of some apps (e.g. Element since we talked about this one), so they might not know that they could have push notifications using the PlayStore version.

This is for instance part of the reasons Signal is not available in F-Droid, because they definitively did not want to push an instant messenger without push notifications (nowadays they are other reasons since they implemented a fallback, but still).

Also from a privacy standpoint you don’t gain a lot by having it from F-Droid rather than the PlayStore, while having a non-FCM version would be a great win.

Finally and foremost, if developers have time to spent on this, I would rather see them implement UnifiedPush than polling: since work is required in both cases, it might as well be in the better direction.

**Comment by IzzySoft** [comment]
> I guess that they will only use it in the F-Droid flavor, not PlayStore one.

I rather guess they'll use it in both for easier maintenance, and have it pre-configured for FCM (which F-Droid users could change if they want).

> Oh ok, I'm personally fine with a polling solution, or even no notifications so long as the app is on F-Droid.

So are most F-Droid users I guess – and that's what I meant by suggesting this as "optional": those who are happy with FCM have that working, those who (like you) are happy with polling just keep it disabled – and all else could choose between Gotify and NoProvider2Push (maybe more in the future). All the devs have to implement for that is: UnifiedPush. A single thing, giving each of the 3 groups what they want.

> Would it be possible to add the app to F-Droid with the Google Play Services requirement removed

That's what build flavors are for – and there are many apps supporting FCM with their Play version having reached F-Droid that way. So it is technically possible – just someone would need to do the work establishing such a flavor. We can kindly ask for this (which we do) – but of course we cannot "demand" it (which we won't, right? :smile:).

> This is for instance part of the reasons Signal is not available in F-Droid

Signal is not on F-Droid because Moxie won't permit that. Moxie wants to stay in absolute control. He said/wrote that more than once. The reasons he mentions sometimes raise questions about whether he knows what he's talking about at all (many of those were bogus or even false, like F-Droid being "insecure", having their signing keys online and similar nonsense), but that's a different thing: with the author explicitly opposing inclusion, F-Droid will not list such an app.

Besides, as we're speaking of a mail app here: as I wrote before, the mail server itself is capable of push notifications. It's called [IMAP IDLE](https://en.wikipedia.org/wiki/IMAP_IDLE) and supported since IMAP4:

> The IDLE feature allows IMAP email users to immediately receive any mailbox changes without having to undertake any action such as clicking on a refresh button, or having the email client automatically and repeatedly ask the server for new messages.

Then there's also [Push-IMAP](https://en.wikipedia.org/wiki/Push-IMAP) with similar support. I'm just not sure if ProtonMail uses IMAP "behind the scenes".

**Comment by ArchangeGabriel** [comment]
> > I guess that they will only use it in the F-Droid flavor, not PlayStore one.
> 
> I rather guess they'll use it in both for easier maintenance, and have it pre-configured for FCM (which F-Droid users could change if they want).

From what I understood of UnifiedPush, if you want to use FCM you still have to install a side component (https://unifiedpush.org/users/distributors/fcm/). For the PlayStore version, that would be a huge friction, so very not likely. But that could be me misunderstanding this case (using UnifiedPush with FCM). It would be nice indeed if UnifiedPush could actually detect GMS/µG with GSF being available and then use them without requiring a distributor (but of course prioritize a distributor if any is available, because an user might have both in which case the distributor is their actual choice), and actually this seems to me like a requirement for wide adoption in the PlayStore (it’s OK to ask F-Droid users to install a distributor, but no developer will want that friction in the PlayStore).

> > This is for instance part of the reasons Signal is not available in F-Droid
> 
> Signal is not on F-Droid because Moxie won't permit that. Moxie wants to stay in absolute control. He said/wrote that more than once. The reasons he mentions sometimes raise questions about whether he knows what he's talking about at all (many of those were bogus or even false, like F-Droid being "insecure", having their signing keys online and similar nonsense), but that's a different thing: with the author explicitly opposing inclusion, F-Droid will not list such an app.

I said “part of the reasons”. Indeed in the end it all comes down to Moxie not wanting that, but amongst the arguments he initially used to justify so (before all of them being shown as non-sensical and then moving to plain “I don’t want that”) was that push would be missing. But then eventually when he came with the WS fallback, he had other reasons and so on.

> Besides, as we're speaking of a mail app here: as I wrote before, the mail server itself is capable of push notifications. It's called [IMAP IDLE](https://en.wikipedia.org/wiki/IMAP_IDLE) and supported since IMAP4:
> 
> > The IDLE feature allows IMAP email users to immediately receive any mailbox changes without having to undertake any action such as clicking on a refresh button, or having the email client automatically and repeatedly ask the server for new messages.

However IMAP IDLE has been shown to be incredibly tough to implement correctly on modern Android: https://github.com/k9mail/k-9/issues/4258

> Then there's also [Push-IMAP](https://en.wikipedia.org/wiki/Push-IMAP) with similar support. I'm just not sure if ProtonMail uses IMAP "behind the scenes".

There is also JMAP nowadays that does a better job than IMAP, which would be a great solution especially in an environment where ProtonMail controls both the server and the clients (maybe they already use that?). But then it turns back to system push instead of IMAP IDLE.

**Comment by julianfairfax** [comment]
> > Would it be possible to add the app to F-Droid with the Google Play Services requirement removed
> 
> That's what build flavors are for – and there are many apps supporting FCM with their Play version having reached F-Droid that way. So it is technically possible – just someone would need to do the work establishing such a flavor. We can kindly ask for this (which we do) – but of course we cannot "demand" it (which we won't, right? smile).

I kindly request...
> > This is for instance part of the reasons Signal is not available in F-Droid
> 
> Signal is not on F-Droid because Moxie won't permit that. Moxie wants to stay in absolute control. He said/wrote that more than once. The reasons he mentions sometimes raise questions about whether he knows what he's talking about at all (many of those were bogus or even false, like F-Droid being "insecure", having their signing keys online and similar nonsense), but that's a different thing: with the author explicitly opposing inclusion, F-Droid will not list such an app.

This is also the reason perfectly working forks like Signal-FOSS aren't in the main repo, he won't allow them.

**Comment by IzzySoft** [comment]
> This is also the reason perfectly working forks like Signal-FOSS aren't in the main repo, he won't allow them.

He especially won't allow them connecting to his servers. While he cannot prevent forks (after all, it's FOSS with a proper license), he can simply block them at the server end – and I've heard he did in the past. Which effectively renders forks moot unless you want to run your own server, too.

But let's stop that sideline here to not distract from the main topic :wink:

> if you want to use FCM you still have to install a side component

See? I wrote I'm not a dev, so I can just guess. And sometimes guesses are a bit off. So it might well be you are correct here and I am wrong. We'll never know before being told…

**Comment by proletarius101** [comment]
>  if you want to use FCM you still have to install a side component

No, if you have Google Play and GMS.

> https://unifiedpush.org/developers/embedded_fcm/
> You may want to embedded a FCM distributor to make notifications invisible for many users. Embedding a FCM Distributor is a solution : if the user doesn’t have another distributor installed, this one will be used.

**Comment by ArchangeGabriel** [comment]
Awesome, thanks for pointing it. So this is definitively the best solution!

**Comment by julianfairfax** [comment]
Well it seems this would be the perfect solution then? In the meantime it'd still be nice to have a version without notifications on F-Droid.

> You have to understand that such applications without push notifications might be seen has bad UX for users (delays in retrieving content and/or increase battery consumption), something developers don’t want. Even if you would be fine with it, the availability of the app might bring other users in trying it and find it deceptive, meaning they will turn away from it and likely won’t ever know (and thus retry) when push notifications will be available.
>
> Also not everyone understand that this difference exists between F-Droid and PlayStore version of some apps (e.g. Element since we talked about this one), so they might not know that they could have push notifications using the PlayStore version.

This version that wouldn't have notifications could a pop-up letting you know you need to install the version from the Google Play Store for notifications and that the developers are working on the issue. A pop-up for Google Play Services already exists in the app so that could just be edited a bit and the message could be added to the app's description on F-Droid and whenever an F-Droid link is mentioned on their site?

**Comment by proletarius101** [comment]
Actually I've made some patches to remove the proprietary libraries, which could be used to do the naive F-droid listing: https://gitlab.com/secure-system/proton/proton-mail-android/-/jobs/1224042042/artifacts/download?file_type=archive

**Comment by julianfairfax** [comment]
> Actually I've made some patches to remove the proprietary libraries, which could be used to do the naive F-droid listing: https://gitlab.com/secure-system/proton/proton-mail-android/-/jobs/1224042042/artifacts/download?file_type=archive

So this build has no proprietary libraries? You could publish this on an/your F-Droid repo if ProtonMail would rather wait until they have their own implementation down before they publish it on the main repo?

**Comment by julianfairfax** [comment]
I hope this isn't taken the wrong way. After all, you've been incredibly kind to make this app free and open source, but F-Droid support, and more specifically, support for notifications without using Google Play Services is long overdue and one of the most important features you're missing when compared to your competition. Especially knowing that the users you're targeting with this are those who are privacy-conscious and do not want to use Google services on their device. This is literally your #1 issue (on this repository).

I know the app is available as an APK from your website, and on third-party F-Droid repos, but that doesn't solve the larger problem of the Google Play Services being included in the APK, and that being only way to get notifications, which are an important feature.

I'm currently using your service for free, which I'm really thankful is even an option, but if anything would incentivize me to get a paid plan, sponsor, and/or donate to this project, it would be this. It's time ProtonMail be added to F-Droid, and be free from Google Play Services. Thank you for the incredible work you've done on this project, and for all the future work you will do!

**Comment by jpellegrini** [comment]
> Especially knowing that the users you're targeting with this are those who are privacy-conscious and do not want to use Google services on their device

I would love to use an email app that does _NOT_ show any notifications at all. For privacy reasons. I know that it is possible to disable notifications, but what I mean is that I wouldn't miss the feature at all, and as far as I can tell, most people that I know who care about privacy wouldn't either.

**Comment by licaon-kter** [comment]
@jpellegrini it's not about the notification, it's about getting messages as they are received (if you wanna get notified or not that's your thing)

**Comment by jpellegrini** [comment]
> @jpellegrini it's not about the notification, it's about getting messages as they are received (if you wanna get notified or not that's your thing)

@licaon-kter I thought it was about using Google Play Services for notifications (?)

**Comment by julianfairfax** [comment]
> > Especially knowing that the users you're targeting with this are those who are privacy-conscious and do not want to use Google services on their device
> 
> I would love to use an email app that does _NOT_ show any notifications at all. For privacy reasons. I know that it is possible to disable notifications, but what I mean is that I wouldn't miss the feature at all, and as far as I can tell, most people that I know who care about privacy wouldn't either.

Well that should be an option of course but I want to know when I'm getting an email in case it's important. And I think most people do. But for what it's worth, ProtonMail has said that they're only going to add it to F-Droid once they work out notifications, so I don't think they're going to change their minds about that being a required feature anyway.

**Comment by IzzySoft** [comment]
As for Push service: could we push for [UnifiedPush](https://unifiedpush.org/) – now that with Gotify-UP there's a push provider available at F-Droid? Implementing UnifiedPush would mean giving the user the choice of the backend to be used:

* no UnifiedPush available: no push (for those who love the "slow pace")
* want that FCM? Can be used (if I read that right, there's a simple bridge to just hand it over to GMS if installed).
* have another UP-client (like GotifyUP) installed and configured? Can even be used automatically when detected.

As UP is fully F/LOSS, it would be accepted at F-Droid. And same as with FCM: regardless how many apps are using it, it just requires a single connection. Currently, there are less than a handful apps supporting it (like Fedilab in its latest beta). Maybe ProtonMail can be challenged to "front up" that list?

**Comment by jpellegrini** [comment]
> Protonmail crippled IMAP, no IDLE being used, so it needs either a background service to periodically check for new mails or a push service

I suppose having the user press a button to retrieve the email would not be an option. Right?
Because I'd *absolutely* do that.

**Comment by licaon-kter** [comment]
@jpellegrini when you open the app and when you pull-to-refresh it updates, the issue is about the app being in background or stopped

**Comment by jpellegrini** [comment]
> @jpellegrini when you open the app and when you pull-to-refresh it updates, the issue is about the app being in background or stopped

Yes -- that's what I meant. I'd be happy using the app in foreground only and not having it fetch or do anything in the background. But I think this is really not usual (people won't want that).

**Comment by julianfairfax** [comment]
Not sure how to reach the developers themselves but are any of you working on *any* of these options? Because if so it would be useless to debate which one to use. And if not, you should take a look at what's being discussed. It's important we know what, if anything, has been done so we discuss stuff that's useful.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
+1 Would really love to see this implemented. I hope that the complications needed to solve this wont take 5+ years of development.

**Comment by perezjasonr** [comment]
+1 would love to see this happen

**Comment by Flam3z** [comment]
Has a dev stated a reasonas to why it has not been officially added yet?

**Comment by jahway603** [comment]
This issue was open over a year ago. This needs to be a priority. I don't install apps from the g00g play store and I know I'm not the only one. Please fix this.

**Comment by julianfairfax** [comment]
With the release of a new version of the Android app to the Google Play Store beta channel today, I was excited at the prospect that we'd finally be getting the open source push notifications implementation and F-Droid release we've been waiting for.

Unfortunately, I soon discovered all we got was a much needed redesign, but no functional changes, and still a supposedly privacy-respecting app fully reliant on Google for notifications and distribution.

This disappointed me. I was hoping today would finally be the day. We need functional improvements more than visual changes, and I think it's about time ProtonMail becomes free from Google. Please let us know when we can see this functionality being added.

Sincerely,
A ProtonMail user

**Comment by julianfairfax** [comment]
> With the release of a new version of the Android app to the Google Play Store beta channel today, I was excited at the prospect that we'd finally be getting the open source push notifications implementation and F-Droid release we've been waiting for.
> 
> Unfortunately, I soon discovered all we got was a much needed redesign, but no functional changes, and still a supposedly privacy-respecting app fully reliant on Google for notifications and distribution.
> 
> This disappointed me. I was hoping today would finally be the day. We need functional improvements more than visual changes, and I think it's about time ProtonMail becomes free from Google. Please let us know when we can see this functionality being added.
> 
> Sincerely, A ProtonMail user

I've decided at this point to just stop using Google Play Services and not receive ProtonMail notifications. I read on https://protonmail.com/blog/engineering-team-2021-update/ that you basically aren't going to update the app any time soon, and I find this really disappointing.

For a service that claims to respect the user's privacy above all else, requiring that we use Google Play Services for basic features such as notifications, is far from being in line with your own promises. I hope that one day in the near future this will be supported, but for now I'm just going to have to live without it.

**Comment by IzzySoft** [comment]
> For a service that claims to respect the user's privacy above all else, requiring that we use Google Play Services for basic features such as notifications, is far from being in line with your own promises.

That's a hard but fully justified argument. All your priacy statements look at least slightly tainted to me because of that. Don't get me wrong, I value very much what you're doing – but this sheds a light of doubt, as Julian correctly points out. You should take that very serious at ProtonMail and make it a priority issue; you cannot fight the devils with a "count of devils" – Google isn't exactly standing for privacy.

That said, let me once more point out that there ARE **GOOD** alternatives. Again, I highly recommend @UnifiedPush – which not only would give the users a choice, but even allow you to have FCM as fallback for your Google Play builds if you like. It supports many "back-ends" meanwhile (ntfy, Gotify, Nextcloud to name some) which are already available via F-Droid (or soon will be, in the case of ntfy). You can find some more details in my list for [Push Clients/Manager](https://android.izzysoft.de/applists/category/named/network_push).

**Comment by julianfairfax** [comment]
> That said, let me once more point out that there ARE **GOOD** alternatives. Again, I highly recommend @UnifiedPush – which not only would give the users a choice, but even allow you to have FCM as fallback for your Google Play builds if you like. It supports many "back-ends" meanwhile (ntfy, Gotify, Nextcloud to name some) which are already available via F-Droid (or soon will be, in the case of ntfy). You can find some more details in my list for [Push Clients/Manager](https://android.izzysoft.de/applists/category/named/network_push).

I think they have already developed their own system, or have at least claimed to have been testing one, but like the rest of the stuff they develop, it's not going to even be released in beta for ages

**Comment by julianfairfax** [comment]
It's just sad to see this kind of issue still open with ProtonMail when Tutanota has had this feature for over 2 years: https://f-droid.org/en/2018/09/03/replacing-gcm-in-tutanota.html

**Comment by polyzen** [comment]
Please stop repeating yourselves and spamming our inboxes, the point has been made numerous times.

**Comment by jeremiah-miller** [comment]
Proton already hosts the app, why can't they just make a custom fdroid repo for it?

**Comment by davidbtc2009** [comment]
Only 3 more years to go before we get unifiedpush. This was opened in 2020 and its 2022 now. Soon®

**Comment by licaon-kter** [comment]
@davidbtc2009 that was never a goal

**Comment by licaon-kter** [comment]
Will the "[new Android app](https://crowdin.com/project/protonmail/en#/Proton%20Mail/Proton%20Mail%20Android%20new%20app%20(wip))" be build from ground up to be more dependant on ~~monopolistic~~Google or not?

**Comment by IzzySoft** [comment]
@marinomeneghel with the APK size soon reaching twice what my repo permits per app, and the app still not ready for F-Droid: may I kindly ask for per-ABI builds to be able of hosting it further? Let me know then if I shall pick the armeabi or arm64 variant, and I'll switch over. Thanks in advance!
